### ISSUE TITLE
Crossbow holdtype cwalk to cidle blend animation problem

### ISSUE THREAD

**Issue by TotallyARussian** [Facepunch/garrysmod-issues#5841]
The player animations for the crossbow holdtype seem to not be working quite right. More specifically the crouch walking animation, there seems to be a mishap with blending? Basically if you crouch walk with alt, it seems to play a mix of the normal non-crouch idle animation and the cwalk animation instead of the crouch idle animation, and if you look closely you can see that when you stop crouch walking, the animation changes from cwalk_crossbow to idle_crossbow for a second before the player velocity slows down to 0, after which it uses the correct cidle_crossbow.

Steps to reproduce:
1. Equip the crossbow
2. Hold the crouch and walk button
3. Move forward while holding the crouch and walk buttons

You can even compare the animation by switching to different weapons from the crossbow while crouch walking with alt.

I'm pretty sure this started after that one update that changed the crouching animations a while ago,

**Comment by robotboy655** [comment]
I am not sure which update you are referring to, considering the animation files for players were not changed in the past 10 years.

**Comment by TotallyARussian** [comment]
nvm i found it, it's this one https://gmod.facepunch.com/news/may-2018-update
 "Improved 3rd person crouching animations to better represent what happens in 1st person"

**Comment by robotboy655** [comment]
https://github.com/Facepunch/garrysmod/commit/25bb767629e9a2c5a4e41e5b539cbf641a80683a

this would be that change, does undoing this change fix the issue you are having? You can do this in your own game files to test this.

**Comment by TotallyARussian** [comment]
no, doesn't seem to change anything.

**Comment by TotallyARussian** [comment]
yeah, this seems to be a problem with the actual cwalk_crossbow sequence, you can even check it in the half life model viewer, it uses the idle animation as a base instead of cidle for some reason

**Comment by TotallyARussian** [comment]
also, i did some more digging around with animations and found that the cwalk_passive sequence also has the exact same issue

**Comment by TotallyARussian** [comment]
so, not sure if this is of any use, but, i have been editing some other player model animations for my player model that i was making, i looked at the cwalk sequence for the crossbow in blender and i checked the .qc file of the animation library model, the problem part of the cwalk_crossbow issue is the "a_CrouchWalking_cwalk_crossbow_C.smd" animation, and since it's a stationary animation, couldn't we just make the .qc file point to the "cidle_crossbow.smd" for the $animation part?

so instead of this:
$animation "a_CrouchWalking_cwalk_crossbow_C" "anims\a_CrouchWalking_cwalk_crossbow_C.smd" {
	fps 30
	loop
	ikrule "rhand" release contact 0 fakeorigin 0 0 0 fakerotate 0 0 0 floor 0 height 0 radius 0 range 0 0 29 29 target 0
	ikrule "lhand" release contact 0 fakeorigin 0 0 0 fakerotate 0 0 0 floor 0 height 0 radius 0 range 0 0 29 29 target 1
	ikrule "rfoot" release contact 0 fakeorigin 0 0 0 fakerotate 0 0 0 floor 0 height 0 radius 0 range 0 0 29 29 target 2
	ikrule "lfoot" release contact 0 fakeorigin 0 0 0 fakerotate 0 0 0 floor 0 height 0 radius 0 range 0 0 29 29 target 3
}

it's this:
$animation "a_CrouchWalking_cwalk_crossbow_C" "anims\cidle_crossbow.smd" {
	fps 30
	loop
	ikrule "rhand" release contact 0 fakeorigin 0 0 0 fakerotate 0 0 0 floor 0 height 0 radius 0 range 0 0 29 29 target 0
	ikrule "lhand" release contact 0 fakeorigin 0 0 0 fakerotate 0 0 0 floor 0 height 0 radius 0 range 0 0 29 29 target 1
	ikrule "rfoot" release contact 0 fakeorigin 0 0 0 fakerotate 0 0 0 floor 0 height 0 radius 0 range 0 0 29 29 target 2
	ikrule "lfoot" release contact 0 fakeorigin 0 0 0 fakerotate 0 0 0 floor 0 height 0 radius 0 range 0 0 29 29 target 3
}
so in theory this should make it use the correct "cidle_crossbow" animation for when the "cwalk_crossbow" sequence has 0's on move_x and move_y poseparameters.

**Comment by TotallyARussian** [comment]
i've also discovered that the sit_revolver animation also doesn't exist for some reason, the game simply uses the normal sit animation, sit_pistol could work for it, and the sit_duel animation is also broken, the upper body is bent down and to the left with default aim angles, as if aiming downwards and to the left. there's also 3 typos with the dual wield holdtype, with the swimming and sit animations using the word "duel" while the rest of the animations use the correct "dual" word. and i have decided to rename the issue to a more appropriate name, since there's more issues than the intial cwalk_crossbow and passive animation problems, i offer my deepest apologies if this causes annoyance, thank you.

I'm going to make a small list here to combine all the current issues with animations for easier understanding:

1. cwalk_crossbow - incorrect base pose (should be cidle_crossbow not idle_crossbow)
2. cidle_passive - doesn't reflect the crouching state very well (player leans forward instead of crouching), cwalk_passive will need to be changed too.
3. sit_duel - aimlayer issues? player aims down-left by default.
4. sit_duel, swim_idle_duel, swimming_duel typos, should be "dual" like the rest of the animations.
